app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,169 @@
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_month
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_number
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_save
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_year
+
+@Composable
+fun AddCreditCardDialog(
+    onCreateCard:(CreditCardInfo) -> Unit,
+    onDismissRequest: () -> Unit,
+) {
+    var name by remember { mutableStateOf("") }
+    var number by remember { mutableStateOf("") }
+    var expirationMonth by remember { mutableStateOf("") }
+    var expirationYear by remember { mutableStateOf("") }
+
+    Dialog(onDismissRequest = onDismissRequest) {
+        Box(modifier = Modifier
+            .clip(RoundedCornerShape(16.dp))
+            .pointerInput(Unit) { detectTapGestures { } }
+            .shadow(8.dp, shape = RoundedCornerShape(16.dp))
+            .background(
+                MaterialTheme.colorScheme.surface
+            ),
+            contentAlignment = Alignment.Center
+        )  {
+            Box(modifier = Modifier
+                .clip(RectangleShape)
+                .fillMaxWidth()
+                .align(Alignment.Center)
+                .background(Color.White)
+            ) {
+                Column {
+                    CreditCardUI(name, number, expirationMonth, expirationYear)
+
+                    TextField(
+                        value = name,
+                        onValueChange = { name = it },
+                        label = { Text(add_credit_card_full_name) },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    TextField(
+                        value = number,
+                        onValueChange = { number = it.take(16) },
+                        label = { Text(add_credit_card_number) },
+                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(dimen_16_dp),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        TextField(
```

Comment:
![Screen Shot 2024-01-27 at 12 33 59 PM](https://github.com/em4n0101/LayoutBasic/assets/16028757/360123b7-e9d1-42d6-a980-b5b6dcb0b1c5)
 still, we need here to add some limitations...for month and year